packages/hoppscotch-common/src/helpers/import-export/import/openapi.ts

```diff
@@ -514,6 +514,32 @@ const parseOpenAPIAuth = (
     ? parseOpenAPIV3Auth(doc as OpenAPIV3.Document | OpenAPIV31.Document, op)
     : parseOpenAPIV2Auth(doc as OpenAPIV2.Document, op)
 
+const parseOpenAPIUrl = (
+  doc: OpenAPI.Document | OpenAPIV2.Document | OpenAPIV3.Document
+): string => {
+  /**
+   * OpenAPI V2 has version as a string in the document's swagger property.
+   * And host and basePath are in the document's host and basePath properties.
+   * Relevant v2 reference: https://swagger.io/specification/v2/#:~:text=to%20be%20obscured.-,Schema,-Swagger%20Object
+   **/
+
+  if (objectHasProperty(doc, "swagger")) {
+    return `${doc.host}${doc.basePath}`
+  }
+
+  /**
+   * OpenAPI V3 has version as a string in the document's openapi property.
+   * And host and basePath are in the document's servers property.
+   * Relevant v3 reference: https://swagger.io/specification/#server-object
+   **/
+  if (objectHasProperty(doc, "servers")) {
+    return doc.servers?.[0].url ?? ""
```

Comment:
Revert this to `<<baseUrl>>` as well since we don't know the base url if this is not given.

Reply:
Resolved in 3d183dc